core/arch/x86/x86.asm

```diff
@@ -796,7 +796,7 @@ GLOBAL_LABEL(global_do_syscall_sygate_int:)
  */
         DECLARE_FUNC(global_do_syscall_sysenter)
 GLOBAL_LABEL(global_do_syscall_sysenter:)
-#if defined(X64) && defined(WINDOWS)
+#if defined(X64)
```

Comment:
Re: whether the kernel will have the SYSENTER MSR set up: I think it would be better to crash at the gateway, which should be not too hard to debug, than to try SYSCALL with the args messed up and potentially not crash or have visible problems until later downstream.  Maybe put in a comment that there are issues here and it is not tested.

Reply:
> I think A is the simplest solution and is what should have been done originally.

Agreed.

Reply:
(Scratch the MSR not being set up comment: we'd only come here if we saw the app using SYSENTER so we would expect it to work at least for the gateway part.)